### ISSUE TITLE
Fix unfixed violations being unreported

### ISSUE THREAD

**Issue by ChenKaiKe** [stylelint/stylelint#4852]
> Clearly describe the bug

"After I executed the command stylelint XXX -- fix, although stylelint helped me fix some errors, the unrepaired errors were not reported, so I can only use the command stylelint to check again."

> Which rule, if any, is the bug related to?

e.g. `number-leading-zero`

> What code is needed to reproduce the bug?

e.g.

```css
a {
  width: 00.1em;
}
```

> What stylelint configuration is needed to reproduce the bug?

.stylelintrc.js :+1: 

module.exports = {
    "extends": [
        "stylelint-config-standard",
        "stylelint-config-rational-order"
    ],
    "plugins": [
        "stylelint-order",
        "stylelint-scss"
    ],
    "rules": {
        // 缩进
        'indentation': 4,
        // 禁止低优先级的选择器出现在高优先级的选择器之后
        'no-descending-specificity': null,
        // 禁止空源
        'no-empty-source': null,
        // at-rule-no-unknown: 屏蔽一些scss等语法检查
        "at-rule-no-unknown": [
            true,
            {
                "ignoreAtRules": [
                    "mixin",
                    "extend",
                    "content"
                ]
            }
        ],
        "comment-empty-line-before": null,
        "no-invalid-double-slash-comments": null,
        "no-descending-specificity": null,
        "declaration-empty-line-before": null,
        "font-family-no-missing-generic-family-keyword": null,
        "no-duplicate-selectors": null
    }
}

> Which version of stylelint are you using?

  "stylelint": "^13.6.0",
    "stylelint-config-rational-order": "^0.1.2",
    "stylelint-config-standard": "^20.0.0",
    "stylelint-order": "^4.1.0",
    "stylelint-scss": "^3.17.2",
    "stylelint-webpack-plugin": "^2.0.0",

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

cli:
 "lint:style": "stylelint \"{packages,static}/**/*.scss\" && stylelint \"{packages,static}/**/*.less\"",
 "lint-fix:style": "stylelint \"{packages,static}/**/*.scss\" --fix && stylelint \"{packages,static}/**/*.less\" --fix",

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

e.g. "Yes, it's related to SCSS nested properties."

> What did you expect to happen?

After I execute the command stylelint XXX -- fix, stylelint helps me fix some errors and reports those that cannot be fixed.

> What actually happened (e.g. what warnings or errors did you get)?

After I executed the command stylelint XXX -- fix, stylelint only helped me fix some errors, but did not report those that could not be fixed.

```shell
test.css
2:4    Expected a leading zero (number-leading-zero)
```

**Comment by jeddy3** [comment]
@ChenKaiKe Thanks for the report.

Can you provide a minimal reproducible test repository, please? The smallest amount of Less or SCSS code, stylelint configuration and package.json to show the problem. It will be difficult for us to look into the problem without it.

Incidentally, you can use `stylelint "{packages,static}/**/*.{scss,less}"`, rather than two separate commands.

**Comment by ChenKaiKe** [comment]
> amount

Thanks for the help and report。I will provide a minimal reproducible test repository late.

**Comment by ChenKaiKe** [comment]
This is my fault，because i don't have true config
